spec/unit/ec2_ami_list_spec.rb

```diff
@@ -0,0 +1,275 @@
+            'imageLocation'       => "microsoft/Windows_Server-2008-R2-SP1-English-64Bit-WebMatrix_Hosting-2012.06.12",
+            'imageOwnerAlias'     => "microsoft",
+            'name'                => "fedora i64",
+            'imageOwnerId'        => "461346954234",
+            'imageState'          => "available",
+            'imageType'           => "machine",
+            'isPublic'            => true,
+            'productCodes'        => [],
+            'rootDeviceName'      => "/dev/sda1",
+            'rootDeviceType'      => "ebs",
+            'stateReason'         => {},
+            'tagSet'              => {},
+            'virtualizationType'  => "hvm"
+          }],
+        'requestId'     => "ba38c315-f1b4-4822-b336-6309bed6d50c"
+        }
+      )
+    end
+
+    it 'invokes validate!' do
+      allow(ec2_connection).to receive(:describe_images).and_return(@describe_images_format)
+      allow(knife_ec2_ami_list.ui).to receive(:warn)
+      expect(knife_ec2_ami_list).to receive(:validate!)
+      knife_ec2_ami_list.run
+    end
+
+    context 'when region is not specified' do
+      it 'shows warning that default region will be will be used' do
+        knife_ec2_ami_list.config.delete(:region)
+        Chef::Config[:knife].delete(:region)
+        ec2_servers = double()
+        allow(ec2_connection).to receive(:describe_images).and_return(@describe_images_format)
+        allow(knife_ec2_ami_list).to receive(:validate!)
+        expect(knife_ec2_ami_list.ui).to receive(:warn).with("No region was specified in knife.rb or as an argument. The default region, us-east-1, will be used:")
+        knife_ec2_ami_list.run
+      end
+    end
+
+    context 'when --owner is passed' do
+      before do
+        allow(knife_ec2_ami_list.ui).to receive(:warn)
+        allow(knife_ec2_ami_list).to receive(:custom_warnings!)
+      end
+
+      context 'When value for owner is nil' do
+        it 'shows the available AMIs List' do
+          knife_ec2_ami_list.config[:owner] = nil
+          allow(ec2_connection).to receive(:describe_images).and_return(@describe_images_format)
+          expect(knife_ec2_ami_list).to receive(:validate!)
+          output_column = ["AMI ID", "Platform", "Architecture", "Size", "Name"]
+          output_column_count = output_column.length
+          output_column += ["ami-4ace6d23", "windows", "x86_64", "30", "Windows_Server", "ami-4ace6d21", "ubuntu", "i386", "10", "ubuntu", "ami-4ace6d29", "fedora", "x86_64", "8", "fedora"]
+          expect(knife_ec2_ami_list.ui).to receive(:list).with(output_column,:uneven_columns_across, output_column_count)
+          knife_ec2_ami_list.run
+        end
+      end
+
+      context 'When value for owner is self' do
+        it 'shows the private AMIs List' do
+          knife_ec2_ami_list.config[:owner] = self
+          allow(ec2_connection).to receive(:describe_images).and_return(@describe_images_format)
```

Comment:
Since we are mocking the output of `describe_images`, this spec is not able to verify the output based on the `--owner` parameter.
This is how `owner` is passed in the code: `connection.describe_images({"Owner"=>"#{owner}"})`.
Since, `describe_image` is a Fog API, I think we can't validate it's output based on the owner. The only thing that we can verify is that if owner is passes by the user, it should be passed to `describe_image` API. e.g.
```
expect(ec2_connection).to recieve(:describe_images).with({"Owner"=>"<owner>"})
```